README.md

```diff
@@ -25,26 +24,84 @@ Welcome to our source speech repository about [mobx-state-tree](https://github.c
+> Sometimes we don't want fields to be required
+ 1. To do so, we can use the type `maybe`, which sets the field to `null` if it's not defined
+ 2. So now, when we instanciate our powerpuff with an empty snapshot, it works !
+ 3. And if we want to instanciate the model with a valued name, we still can do it 👍
+
+#### [#32](https://github.com/Dedetat/powerpuff-yourself/pull/32/files) - Use more complex types
+> There are lots of available types provided by mobx-state-tree, find the entire list in the [documentation](https://github.com/mobxjs/mobx-state-tree#types-overview)
+ 1. Because a Powerpuff is not only a name but also has feelings, we add the field `mood` to describe her
+ 2. But each Powerpuff has her own personality so let's say she must be into a mood defined in an enumeration
+ 3. If you try to create a Powerpuff with a mood which is not in the enumeration, it fails (Don't try to counterfeit a Powerpuff 💪)
+ 4. Use a mood of the enumeration and now you can instanciate a new Powerpuff !
+
 
 ### Actions
+> This is how we update our model
+
+#### [#33](https://github.com/Dedetat/powerpuff-yourself/pull/33/files) - Updating the model, the natural way
+> As you must be aware, Buttercup is more `aggressive` rather than `happy`. But the powerpuff is now instanciated, the damage is done. The unique solution to repair this mistake is to modify the instance.
+ 1. The natural way would be to set directly the `mood` attribute of the instance
+ 2. As you can guess, it crashes 💥
+
+#### [#34](https://github.com/Dedetat/powerpuff-yourself/pull/34/files) - Updating the model, mobx-state-tree way
+> To mutate the model, mobx-state-tree wants you to create an action. That way the immutability of the state is kept 💖
+ 1. Describe the action within the function `types.actions`
+ 2. Actions take `self` as parameter, it represents the instance itself. They return an object of functions updating the state
+ 3. Here our action is very simple : we set the `mood` field of the instance with the given `mood`
+ 4. Use this action to modify the instance, it works 🎉
+
+
 ### Views
+> There are functions that return values deduced from attributes. This is what we call views.
+
+#### [#35](https://github.com/Dedetat/powerpuff-yourself/pull/35/files) - Create a view
+> Do you know how old is our Powerpuffs now ?
+ 1. The best way to know is to add a `birthday` field to our model and then deduce the age from it 💡
+ 2. Btw, you can assign a value to a field when you create the model 👍
+ 3. Now we can create a view which computes the actual age of our powerpuff. Use `types.views` to describe it
+ 4. Like actions, views take `self` as parameter and return an object of functions computing results
+ 5. Here we write a simple getter to have the current age of the powerpuff and use it like an attribute
+ 6. Calling this view on the instance gives us the current age of the powerpuff 🎉
+
```

Comment:
We miss the part about snapshot that doesn't contain the `age` this can be a note